src/databricks/labs/ucx/source_code/linters/imports.py

```diff
@@ -303,40 +346,32 @@ def _convert_dbutils_notebook_run_to_advice(cls, node: ast.AST) -> Advisory:
         )
 
     @staticmethod
-    def get_dbutils_notebook_run_path_arg(node: ast.Call):
+    def get_dbutils_notebook_run_path_arg(node: Call):
         if len(node.args) > 0:
             return node.args[0]
         arg = next(kw for kw in node.keywords if kw.arg == "path")
         return arg.value if arg is not None else None
 
     @staticmethod
     def list_dbutils_notebook_run_calls(linter: ASTLinter) -> list[NotebookRunCall]:
-        calls = linter.locate(ast.Call, [("run", ast.Attribute), ("notebook", ast.Attribute), ("dbutils", ast.Name)])
+        calls = linter.locate(Call, [("run", Attribute), ("notebook", Attribute), ("dbutils", Name)])
         return [NotebookRunCall(call) for call in calls]
 
-    @staticmethod
-    def list_import_sources(linter: ASTLinter, problem_type: P) -> tuple[list[ImportSource], list[P]]:
+    @classmethod
+    def list_import_sources(cls, linter: ASTLinter, problem_type: P) -> tuple[list[ImportSource], list[P]]:
         problems: list[P] = []
+        sources: list[ImportSource] = []
         try:  # pylint: disable=too-many-try-statements
-            nodes = linter.locate(ast.Import, [])
-            sources = [ImportSource(node, alias.name) for node in nodes for alias in node.names]
-            nodes = linter.locate(ast.ImportFrom, [])
-            sources.extend(ImportSource(node, node.module) for node in nodes)
-            nodes = linter.locate(ast.Call, [("import_module", ast.Attribute), ("importlib", ast.Name)])
-            nodes.extend(linter.locate(ast.Call, [("__import__", ast.Attribute), ("importlib", ast.Name)]))
-            for node in nodes:
-                if isinstance(node.args[0], ast.Constant):
-                    sources.append(ImportSource(node, node.args[0].value))
-                    continue
-                problem = problem_type(
-                    'dependency-not-constant',
-                    "Can't check dependency not provided as a constant",
-                    start_line=node.lineno,
-                    start_col=node.col_offset,
-                    end_line=node.end_lineno or 0,
-                    end_col=node.end_col_offset or 0,
-                )
-                problems.append(problem)
+            nodes = linter.locate(Import, [])
+            for source in cls._make_sources_for_import_nodes(nodes):
+                sources.append(source)
```

Comment:
```suggestion
            sources.extend(cls._make_sources_for_import_nodes(nodes)
```

Reply:
For loops are more readable. For example, right now, with a heavy jetlag, I struggle reading previous version with list compressions